src/main/java/cms/gov/madie/measure/services/TestCaseShiftDatesService.java

```diff
@@ -0,0 +1,282 @@
+import gov.cms.madie.models.cqm.datacriteria.MedicationDispensed;
+import gov.cms.madie.models.cqm.datacriteria.MedicationOrder;
+import gov.cms.madie.models.cqm.datacriteria.Participation;
+import gov.cms.madie.models.cqm.datacriteria.PatientCareExperience;
+import gov.cms.madie.models.cqm.datacriteria.PatientCharacteristic;
+import gov.cms.madie.models.cqm.datacriteria.PatientCharacteristicBirthdate;
+import gov.cms.madie.models.cqm.datacriteria.PatientCharacteristicClinicalTrialParticipant;
+import gov.cms.madie.models.cqm.datacriteria.PatientCharacteristicExpired;
+import gov.cms.madie.models.cqm.datacriteria.PatientCharacteristicPayer;
+import gov.cms.madie.models.cqm.datacriteria.PhysicalExamOrder;
+import gov.cms.madie.models.cqm.datacriteria.PhysicalExamPerformed;
+import gov.cms.madie.models.cqm.datacriteria.PhysicalExamRecommended;
+import gov.cms.madie.models.cqm.datacriteria.ProcedureOrder;
+import gov.cms.madie.models.cqm.datacriteria.ProcedurePerformed;
+import gov.cms.madie.models.cqm.datacriteria.ProcedureRecommended;
+import gov.cms.madie.models.cqm.datacriteria.ProviderCareExperience;
+import gov.cms.madie.models.cqm.datacriteria.SubstanceAdministered;
+import gov.cms.madie.models.cqm.datacriteria.SubstanceOrder;
+import gov.cms.madie.models.cqm.datacriteria.SubstanceRecommended;
+import gov.cms.madie.models.cqm.datacriteria.Symptom;
+import gov.cms.madie.models.cqm.datacriteria.basetypes.DataElement;
+import gov.cms.madie.models.cqm.datacriteria.basetypes.TestCaseJson;
+import gov.cms.madie.models.measure.TestCase;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Service
+public class TestCaseShiftDatesService {
+
+  private final TestCaseService testCaseService;
+
+  private ObjectMapper mapper = new ObjectMapper().registerModule(new JavaTimeModule());
+
+  @Autowired
+  public TestCaseShiftDatesService(TestCaseService testCaseService) {
+    this.testCaseService = testCaseService;
+  }
+
+  public TestCase shiftTestCaseDates(
+      String measureId, String testCaseId, int shifted, String username, String accessToken) {
+
+    TestCase testCase = null;
+    List<TestCase> testCases = testCaseService.findTestCasesByMeasureId(measureId);
+    if (CollectionUtils.isEmpty(testCases)) {
+      throw new ResourceNotFoundException("TestCase", measureId);
+    } else {
+      Optional<TestCase> existingOpt =
+          testCases.stream().filter(tc -> tc.getId().equals(testCaseId)).findFirst();
+      if (existingOpt.isEmpty()) {
+        throw new ResourceNotFoundException("TestCase", measureId);
+      }
+      testCase = existingOpt.get();
+    }
+
+    shiftDatesForTestCase(testCase, shifted, measureId, username, accessToken);
+    testCaseService.updateTestCase(testCase, measureId, username, accessToken);
+
+    return testCase;
+  }
+
+  protected TestCase shiftDatesForTestCase(
+      TestCase testCase, int shifted, String measureId, String username, String accessToken) {
```

Comment:
I think a couple of these parameters are unused

Reply:
@adongare Yeah it was due to a code refactor, I removed the unused parameters, thanks!